### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by LLMsKsenia**
Changes:
1. I broke down LOOCV into two functions: sample and subject
2. Edited LOOCV code for sample to remove the 'fold_...' lists and np.mean(fold_...) as they were no longer needed since the metrics calculated outside of the names loop on the final lists of prediction and tests and return one value
3. Wrote subject LOOCV - store modes for prediction and test for each subject within names loop. This results in two lists with length of 18 or more (in case multiple modes are returned). If multiple modes for prediction returned (both AD and HC) than additional test value is added to the test_mode list to match the length of prediction_mode list. Then outside of loop calculate the metrics based on these mode lists
4. Added a function that returns multiple modes
5. Re-wrote functions to calculate mean from 5 runs of both Sample and Subject LOOCV

Note: metrics for Subject LOOCV are of course lower and often some of them have the same values (e.g. [[0.43333333 0.23161765 0.43333333 0.30174359 0.43333333]])

Outstanding:
In Subject LOOCV I am still getting a warning about ill-defined metrics which blows my mind as they are calculated outside of the names loop and since there are healthy subjects in the study there are always both AD and HC labels in test_mode list. So, I don't know what causes this warning. Will try to figure it out

### CODE REVIEW

File: RandomForest_v9.ipynb [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+    "        X = df_data[featureset].values\n",
+    "        y_label = df_data['group_label_id'].values\n",
+    "        years = df_data['age_at_record'].values\n",
+    "        years_diag = df_data['year_diag'].values\n",
+    "        sub_id = df_data['subject_id'].values\n",
+    "        X = StandardScaler().fit_transform(X)\n",
+    "        sel = VarianceThreshold(threshold=(.99 * (1 - .99)))\n",
+    "        X = sel.fit_transform(X)\n",
+    "        ind_train = [~df_data['firstname'].isin([name])]\n",
+    "        ind_test = [df_data['firstname'].isin([name])] \n",
```
Comment (jeknov) [inline RandomForest_v9.ipynb]: Lines 172 and 173 are the same as 160-161. Delete either the first ones or the second.

File: RandomForest_v9.ipynb [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+    "        ind_train = [~df_data['firstname'].isin([name])]\n",
+    "        ind_test = [df_data['firstname'].isin([name])]\n",
+    "        df_data = df_data.sort_values('age_at_record')\n",
+    "        #for every train-test split, obtain means only from training HC group (so that model is completely agnostic to labels)\n",
+    "        X = df_data[featureset].values\n",
+    "        y_label = df_data['group_label_id'].values\n",
+    "        years = df_data['age_at_record'].values\n",
+    "        years_diag = df_data['year_diag'].values\n",
+    "        sub_id = df_data['subject_id'].values\n",
+    "        X = StandardScaler().fit_transform(X)\n",
```
Comment (jeknov) [inline RandomForest_v9.ipynb:166]: Lines 166-168 are not necessary, not used later in the code

File: RandomForest_v9.ipynb [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+    "        sel = VarianceThreshold(threshold=(.99 * (1 - .99)))\n",
+    "        X = sel.fit_transform(X)\n",
+    "        ind_train = [~df_data['firstname'].isin([name])]\n",
+    "        ind_test = [df_data['firstname'].isin([name])] \n",
```
Comment (jeknov) [inline RandomForest_v9.ipynb]: Same as in the previous function - ind_train and ind_test defined twice

File: RandomForest_v9.ipynb [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+    "        df_data = df_data[df_data.firstname.isin(subject_names)]\n",
+    "        ind_train = [~df_data['firstname'].isin([name])]\n",
+    "        ind_test = [df_data['firstname'].isin([name])]\n",
+    "        df_data = df_data.sort_values('age_at_record')\n",
+    "        #for every train-test split, obtain means only from training HC group (so that model is completely agnostic to labels)\n",
+    "        X = df_data[featureset].values\n",
+    "        y_label = df_data['group_label_id'].values\n",
+    "        years = df_data['age_at_record'].values\n",
+    "        years_diag = df_data['year_diag'].values\n",
+    "        sub_id = df_data['subject_id'].values\n",
```
Comment (jeknov) [inline RandomForest_v9.ipynb]: years, years_diag and sub_id are not used anywhere

**Review by jeknov — COMMENTED** [review]
All I can suggest wrt warnings:
`import warnings`
`warnings.filterwarnings("ignore")`  :)
